sdk/storage/azblob/blob/client_test.go

```diff
@@ -191,6 +191,67 @@ func waitForCopy(_require *require.Assertions, copyBlobClient *blockblob.Client,
 	}
 }
 
+func (s *BlobUnrecordedTestsSuite) TestCopyBlockBlobFromUrlSourceContentMD5() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	if err != nil {
+		s.Fail("Unable to fetch service client because " + err.Error())
+	}
+
+	containerName := testcommon.GenerateContainerName(testName)
+	containerClient := testcommon.CreateNewContainer(context.Background(), _require, containerName, svcClient)
+	defer testcommon.DeleteContainer(context.Background(), _require, containerClient)
+
+	const contentSize = 8 * 1024 // 8 KB
+	content := make([]byte, contentSize)
+	contentMD5 := md5.Sum(content)
+	body := bytes.NewReader(content)
+
+	srcBlob := containerClient.NewBlockBlobClient("srcblob")
+	destBlob := containerClient.NewBlockBlobClient("destblob")
+
+	// Prepare source bbClient for copy.
+	_, err = srcBlob.Upload(context.Background(), streaming.NopCloser(body), nil)
+	_require.Nil(err)
+
+	expiryTime, err := time.Parse(time.UnixDate, "Fri Jun 11 20:00:00 UTC 2049")
+	_require.Nil(err)
+
+	credential, err := testcommon.GetGenericSharedKeyCredential(testcommon.TestAccountDefault)
+	if err != nil {
+		s.T().Fatal("Couldn't fetch credential because " + err.Error())
+	}
+
+	// Get source blob url with SAS for StageFromURL.
+	sasQueryParams, err := sas.AccountSignatureValues{
+		Protocol:      sas.ProtocolHTTPS,
+		ExpiryTime:    expiryTime,
+		Permissions:   to.Ptr(sas.AccountPermissions{Read: true, List: true}).String(),
+		ResourceTypes: to.Ptr(sas.AccountResourceTypes{Container: true, Object: true}).String(),
+	}.SignWithSharedKey(credential)
+	_require.Nil(err)
+
+	srcBlobParts, _ := blob.ParseURL(srcBlob.URL())
+	srcBlobParts.SAS = sasQueryParams
+	srcBlobURLWithSAS := srcBlobParts.String()
+
+	// Invoke CopyFromURL.
+	sourceContentMD5 := contentMD5[:]
+	resp, err := destBlob.CopyFromURL(context.Background(), srcBlobURLWithSAS, &blob.CopyFromURLOptions{
+		SourceContentMD5: sourceContentMD5,
```

Comment:
Shouldn't we use the interface type `blob.SourceContentValidationType` here?
Is it that CopyFromURL() only supports `x-ms-source-content-md5` now, so we directly have the `[]byte` type for `SourceContentMD5`.
cc: @seanmcc-msft , @vibhansa-msft

Reply:
@souravgupta-msft, no, I don't think so.  Can you provide a link to this interface?

Because this is an MD5 on source data that the SDK never downloads, it is not possible for the SDK to calculate a checksum.  In addition, there are currently no plans to add CRC-64 support for source checksums in the storage service.

Reply:
This is the `SourceContentValidationType` interface.
https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/blob/constants.go#L202

It has two implementations, `SourceContentValidationTypeCRC64` and `SourceContentValidationTypeMD5`. We do not calculate checksum for source content for this interface.

Reply:
CRC64 is not supported by the service, and there are no plans to do so as far as I know.  I think we should leave this as a byte[].

Reply:
CopyFromURL only accepts `x-ms-source-content-md5` so using the SourceContentValidationType wouldn't work as it supports both CRC64 and MD5. (https://learn.microsoft.com/en-us/rest/api/storageservices/copy-blob-from-url?tabs=azure-ad#request-headers)